### ISSUE TITLE
Better menu context awareness?

### ISSUE THREAD

**Issue by Weikardzaena** [stepmania/stepmania#1174]
When I'm playing on a pad, I find that it's really easy to hold down the back button a little too long when quitting out of a song, and it kicks me to the main menu.  This also happens when I press start to begin a song, and if I hold down the pad a little too long it goes into the settings menu.  I know this is by design, but it seems to me that it would be easier to navigate if at each screen change the game required another press of that same button for it to register.

In other words, if I hold 'back' to quit a song, make it so you have to press 'back' again to go back to the previous menu.  Likewise with the start button.  This isn't such a big deal on keyboard, but I think because you have to put your weight on the buttons for pad, its easier to hold down the button for a bit too long.

Another approach might be to simply change the amount of time between screens.  I'm just throwing ideas out there.

**Comment by kyzentun** [comment]
The pause menu should be used for quitting out of a song instead of holding a button.  I'm going to add a metric to disable the normal backing out logic if the theme has a pause menu, because the pause menu provides a better way of backing out.

Holding start to go into options instead of pressing it twice is only kept around to allow themes to pretend they're DDR.  It's not a good design choice and I'd like to get rid of it.

Adding more time between screens would get in the way of playing the game.

**Comment by kyzentun** [comment]
This commit adds a preference for disabling the "Hold For Options" behavior, to require a second press for options.  Also changes the back button to require a new press on each screen.
It also forces using the pause menu to exit a song instead of holding Back, Select, or Start.

https://github.com/stepmania/stepmania/commit/7bd64c32af226f7a0fffc3bbc76fef042471deff

**Comment by Weikardzaena** [comment]
Awesome!  Glad it was an easy fix.  Thanks for implementing this so quickly :)
